src/rerank/rerank_calculator_ov.cc

```diff
@@ -0,0 +1,315 @@
+
+        auto input_ids = ov::Tensor(ov::element::i64, ov::Shape{batch_size, total_tokens_count_per_batch});
+        auto attention_mask = ov::Tensor(ov::element::i64, ov::Shape{batch_size, total_tokens_count_per_batch});
+
+        // Combine query and document tokens
+        // Schema (tokenizer must be exported without --add_special_tokens flag, we will add it manually)
+        /*
+            BOS_TOKEN  <QUERY TOKENS>  EOS_TOKEN SEP_TOKEN  <DOCUMENT_1 TOKENS>  EOS_TOKEN
+            BOS_TOKEN  <QUERY TOKENS>  EOS_TOKEN SEP_TOKEN  <DOCUMENT_2 TOKENS>  EOS_TOKEN
+            BOS_TOKEN  <QUERY TOKENS>  EOS_TOKEN SEP_TOKEN  <DOCUMENT_3 TOKENS>  EOS_TOKEN
+            BOS_TOKEN  <QUERY TOKENS>  EOS_TOKEN SEP_TOKEN  <DOCUMENT_N TOKENS>  EOS_TOKEN
+        */
+
+        for (size_t i = 0; i < batch_size; i++) {
+            int64_t* input_ids_data = reinterpret_cast<int64_t*>(input_ids.data()) + i * total_tokens_count_per_batch;
+            int64_t* attention_mask_data = reinterpret_cast<int64_t*>(attention_mask.data()) + i * total_tokens_count_per_batch;
+
+            int64_t* out_input_ids_data = reinterpret_cast<int64_t*>(out_input_ids.data()) + i * tokens_count_of_longest_document;
+
+            // Fill input_ids
+            input_ids_data[0] = this->bos_token;
+            std::memcpy(input_ids_data + 1, query_tokens.data(), query_tokens.size() * sizeof(int64_t));
+            input_ids_data[query_tokens.size() + 1] = this->eos_token;
+            input_ids_data[query_tokens.size() + 2] = this->sep_token;
+            std::memcpy(input_ids_data + 1 + query_tokens.size() + 2, out_input_ids_data, tokens_count_of_longest_document * sizeof(int64_t));
+
+            input_ids_data[total_tokens_count_per_batch - 1] = this->pad_token;
+
+            auto it = std::find(out_input_ids_data, out_input_ids_data + tokens_count_of_longest_document, this->pad_token);
+            size_t pad_token_index = (it != out_input_ids_data + tokens_count_of_longest_document) ? std::distance(out_input_ids_data, it) : tokens_count_of_longest_document;
+
+            input_ids_data[1 + query_tokens.size() + 2 + pad_token_index] = this->eos_token;
+
+            // Fill attention_mask
+            std::fill(attention_mask_data, attention_mask_data + total_tokens_count_per_batch, int64_t(0));
+            std::fill(attention_mask_data, attention_mask_data + 1 + query_tokens.size() + 2 + pad_token_index + 1, int64_t(1));
+        }
+
+        return std::make_pair(input_ids, attention_mask);
+    }
+
+    std::vector<float> ComputeScoresUsingRerankModel(ov::Tensor input_ids, ov::Tensor attention_mask, const std::vector<size_t>& chunkMapping, size_t actual_batch_size) const {
+        ModelMetricReporter tmp(nullptr, nullptr, "example_pipeline_name", 1);
+        auto executingStreamIdGuard = std::make_shared<ExecutingStreamIdGuard>(rerank_session->getInferRequestsQueue(), tmp);
+        ov::InferRequest& inferRequest = executingStreamIdGuard->getInferRequest();
```

Comment:
what about the same fixes for embeddings calculator? regarding infer request scope